src/components/MessagesList/MessagesGroup/Message/MessagePart/Poll.vue

```diff
@@ -52,111 +52,94 @@
+			<div class="poll__modal" :class="{'poll__modal--closed' : !isPollOpen }">
+				<!-- Title -->
+				<div class="poll__header">
+					<PollIcon :size="20" />
+					<h2 class="poll__modal-title">
+						{{ pollName }}
+					</h2>
+				</div>
+				<div class="poll__summary">
+					{{ pollSummaryText }}
+				</div>
+
+				<!-- options -->
+				<div v-if="modalPage === 'voting'" class="poll__modal-options">
+					<template v-if="checkboxRadioSwitchType === 'radio'">
+						<NcCheckboxRadioSwitch v-for="(option, index) in options"
+							:key="'radio' + index"
+							:checked.sync="vote"
+							class="poll__option"
+							:value="index.toString()"
+							:type="checkboxRadioSwitchType"
+							name="answerType">
+							{{ option }}
+						</NcCheckboxRadioSwitch>
+					</template>
+					<template v-else>
+						<NcCheckboxRadioSwitch v-for="(option, index) in options"
+							:key="'checkbox' + index"
+							:checked.sync="vote"
+							:value="index.toString()"
+							:type="checkboxRadioSwitchType"
+							name="answerType">
+							{{ option }}
+						</NcCheckboxRadioSwitch>
+					</template>
+				</div>
+
+				<!-- results -->
+				<div v-if="modalPage === 'results'" class="results__options">
+					<div v-for="(option, index) in options"
+						:key="index"
+						class="results__option">
+						<div class="results__option-title">
+							<p> {{ option }} </p>
+							<p class="percentage">
+								{{ getVotePercentage(index) + '%' }}
+							</p>
+						</div>
+						<div v-if="getFilteredDetails(index).length > 0 || selfHasVotedOption(index)"
+							class="results__option__details">
+							<PollVotersDetails v-if="details"
+								:details="getFilteredDetails(index)" />
+							<p v-if="selfHasVotedOption(index)" class="results__option-subtitle">
+								{{ t('spreed', 'You voted for this option') }}
+							</p>
 						</div>
+						<NcProgressBar class="results__option-progress"
+							:value="getVotePercentage(index)"
+							size="medium" />
 					</div>
-					<div v-if="isPollOpen"
-						class="poll__modal-actions">
-						<!-- Vote again-->
-						<NcButton type="secondary"
-							@click="modalPage = 'voting'">
-							{{ t('spreed', 'Change your vote') }}
-						</NcButton>
-						<!-- End poll button-->
-						<NcButton v-if="canEndPoll"
-							type="error"
-							@click="endPoll">
+				</div>
+
+				<div v-if="isPollOpen" class="poll__modal-actions">
+					<!-- Submit vote button-->
+					<NcButton v-if="modalPage === 'voting'"
+						type="primary"
+						:disabled="!canSubmitVote"
+						@click="submitVote">
+						{{ t('spreed', 'Submit vote') }}
+					</NcButton>
+					<!-- Vote again-->
+					<NcButton v-else
+						type="secondary"
+						@click="modalPage = 'voting'">
+						{{ t('spreed', 'Change your vote') }}
+					</NcButton>
+					<!-- End poll button-->
+					<NcActions v-if="canEndPoll"
+						force-menu
+						:container="container">
+						<NcActionButton type="error" @click="endPoll">
```

Comment:
```suggestion
						<NcActionButton @click="endPoll">
```

Nice try, but it doesn't applicable to NcActionButton. Try with a class and deep styles

Reply:
@szaimen does it make sense to provide this prop in vue-library?

Reply:
I will just remove it, why would we need it to be red in a popover ?

Reply:
We usually use red to indicate "Delete sthg". 
1/ End poll  is not like Delete in severity, also it is the only option in the menu for now so it already has its sufficient weight. Adding red would be very weird ( IMO) 
![image](https://github.com/nextcloud/spreed/assets/84044328/6a4864cb-ebed-4431-bf77-104c28a5c872)
2/ If it is usually used for delete actions, from first sight and according to the color, it seems like Delete Poll (IMO)

Anyway, should the text and the icon be red or only the icon ?  
I will also change the icon to sthg like this https://pictogrammers.com/library/mdi/icon/cancel/ cc @szaimen 
Because check mark with red is kinda contradictory

Reply:
> We usually use red to indicate "Delete sthg".
1/ End poll is not like Delete in severity

We use red for actions which are not reversible. Closing a poll is not reversible

Reply:
There is also a warning color we might check. It looks less controversial for me